server/httpin/gateway/oauth2.go

```diff
@@ -3,48 +3,46 @@ package gateway
 import (
 	"net/http"
 
-	"github.com/mattermost/mattermost-plugin-apps/server/proxy"
+	"github.com/mattermost/mattermost-plugin-apps/server/incoming"
 	"github.com/mattermost/mattermost-plugin-apps/utils"
 	"github.com/mattermost/mattermost-plugin-apps/utils/httputils"
 )
 
-func (g *gateway) remoteOAuth2Connect(w http.ResponseWriter, req *http.Request, in proxy.Incoming) {
-	appID := appIDVar(req)
-	log := g.conf.Logger().With("app_id", appID, "acting_user_id", in.ActingUserID)
-
+func (g *gateway) remoteOAuth2Connect(req *incoming.Request, w http.ResponseWriter, r *http.Request) {
```

Comment:
Maybe as a separate PR, but can we embed `*http.Request` into `incoming.Request`? (Maybe along with the payload?)

Reply:
Discussed offline: `incoming.Request` is interdependent  of the protocol used to generate it.